lib/metasearch_index/deleter.rb

```diff
@@ -0,0 +1,55 @@
+module MetasearchIndex
+  module Deleter
+    class V2
+      class MissingArgument < StandardError; end
+      class NotFound < StandardError; end
+      class UnknownError < StandardError; end
+
+      def initialize(id:, data:)
+        @id = id.presence || raise(MissingArgument, "ID and Type must be supplied.")
+        @data = data.presence || raise(MissingArgument, "Not record provided to insert into Elasticsearch.")
+      end
+
+      def delete
+        processor = ElasticsearchProcessor.new
+        processor.delete(self)
+        response = processor.commit
+        validate_response!(response.first)
+      end
+
+      def identifier
+        {
+          _type: _type,
+          _id: base_path,
+        }
+      end
+
+      def validate_response!(response)
```

Comment:
As discussed earlier - there is some logic duplicated between this method and `PublishingEventWorker` that we could potentially extract out, but I haven't thought too much about what that would look like. It would be good to have the core deletion logic, logging, stats, and error reporting implemented once in an index-independent way if there's a nice way of doing that.

Similar situation for the inserter.

Reply:
I agree, have started on this by extracting the code for these new classes to a shared location. will see if this I can include the govukindex code as well.